aws/components/ecs/setup.ftl

```diff
@@ -1611,17 +1611,21 @@
                 [/#list]
             [/#list]
 
-            [@createECSTask
-                id=taskId
-                name=taskName
-                engine=engine
-                containers=containers
-                role=roleId
-                executionRole=executionRoleId
-                networkMode=networkMode
-                dependencies=dependencies
-                fixedName=solution.FixedName
-            /]
+            [#if containers?size > 0 ]
+                [@createECSTask
+                    id=taskId
+                    name=taskName
+                    engine=engine
+                    containers=containers
+                    role=roleId
+                    executionRole=executionRoleId
+                    networkMode=networkMode
+                    dependencies=dependencies
+                    fixedName=solution.FixedName
+                /]
+            [#else]
+                [@fatal message="No container to deploy services or tasks into" /]
```

Comment:
The exception probably needs some more context.

- The containers belong to the task/service not the other way around  
- The message should be a quick message outlining what the problem was 
- The context of the message should include things like taskId 

You also don't need to make creating the task conditional. The createTemplate process will fail as a fatal exception has been added to the log messages. Making the task creation conditional means that we can't look at the fatal template during debugging. 

so might be better to just put the if conditional around the fatal message

Reply:
Cool